cobra/util/solver.py

```diff
@@ -397,9 +402,9 @@ def fix_objective_as_constraint(model, fraction=1, bound=None,
 
 def check_solver_status(status, raise_error=False):
     """Perform standard checks on a solver's status."""
-    if status == optlang.interface.OPTIMAL:
+    if status == OPTIMAL:
         return
-    elif status == optlang.interface.INFEASIBLE and not raise_error:
+    elif (status in has_primals) and not raise_error:
         warn("solver status is '{}'".format(status), UserWarning)
     elif status is None:
```

Comment:
I wonder if we should raise an `OptimizationError` here rather than a `RuntimeError`.